evalml/utils/woodwork_utils.py

```diff
@@ -147,5 +147,6 @@ def downcast_nullable_types(X, ignore_null_cols=True):
     new_ltypes.update(
         {col: "Double" for col in X_int_nullable_cols if col in non_null_columns},
     )
-    X.ww.init(schema=original_X_schema, logical_types=new_ltypes)
+    if new_ltypes:
```

Comment:
This was causing some weirdness where even though `new_ltypes` was empty - `BooleanNullable` columns would be re-inferred as a uknown string. @ParthivNaresh any idea why this might come up?

Reply:
The string uknown fix is actually in lines [147](https://github.com/alteryx/evalml/pull/3686/files#diff-19da4cc2ff7fdbf98a854e6d8a0ce26496df464ff0f2f0d8b4d470d1f54ace31R147-R153). I'm going to leave this if statement in as I don't think we need to go through an reinit if there's nothing to change!

Reply:
@jeremyliweishih I think you should be good to use `.ww.set_types` here instead of reinitializing Woodwork. Furthermore, I think you might be able to simplify this code some by doing so. You should be able to drop all the code related to saving the original schema at the start, and use the `ww.select` method to get the boolean nullable and integer nullable columns directly.